src/netCLISample/Program.cs

```diff
@@ -8,49 +8,67 @@ namespace netCLISample
 {
     class Program
     {
+        static ILogger<Program> logger;
+
         static int Main(string[] args)
         {
             var services = new ServiceCollection();
             services.AddLogging(builder =>
             {
-                builder.ClearProviders();
-                builder.AddProvider(new MyConsoleLoggerProvider());
+                builder.SetMinimumLevel(LogLevel.Debug);
+                builder.AddConsole();
             });
-            services.ConfigureCLFlow(config =>
+            services.ConfigureCLFlow(mapping =>
             {
-                config.Map<MoveCommand, MoveCommandHandler>()
+                mapping.Map<MoveCommand, MoveCommandHandler>()
                       .Map<HelpCommand, HelpCommandHandler>();
 
             });
 
-            var provider = services.BuildServiceProvider();
+            using( var provider = services.BuildServiceProvider())
+            {
+                logger = provider.GetRequiredService<ILogger<Program>>();
 
-            return CLFlow.Create().UseServiceProvider(provider)
-                                  .UseFallback(ErrorHandling)
-                                  .Run(args);
+                return CLFlow.Create().UseServiceProvider(provider)
+                                      .UseFallback(ErrorHandling)
```

Comment:
Not sure about this fallback, but this may be subject to further improvements.
* What if I want to log my error to eg. database? I would like my fallback function to be able to connect to DB and store some log entries etc.
* In my fallback handler (or should we call it "error handler"?), how can I understand what actually happened? How do I know which command caused the error etc?

Consider below pseudocode for scenario that I'm thinking of:

```
// CLI lib defines this kind of interface
interface ICliErrorHandler
{
    void HandleError(ICommand, ErrorCode, <some other helpful params here?>);
}
```
```
// User can implement this interface - if not, default implementation can be used.
class MyCliErrorHandler : ICliErrorHandler { ... }
```
```
// I can register my imlpementation of ICliErrorHandler interface when configuring CLI
var myHandler = new MyCliErrorHandler(<some dependencies here>);
CLFlow.Create().UseFallback(myHandler);
```

Reply:
Generally you should use your `CommandHandlers` to catch exceptions and log / print related information. `Fallback` method is only for unexpected error and framework related error handling like invalid option etc - that would otherwise prevent executing `Run` method of the handler. I do see value in adding additional information to the `Fallback` method but I don't see it as priority.

Reply:
I would suggest using `ILogger<Handler>` inside `CommandHandlers` just as we do in `HelpCommandHandler`-for variety of reasons `Console.WriteLine` should be avoided.